### PR TITLE
chore: HQM-154: Update to RN 0.72

### PR DESCRIPTION
**Pull request by devkudasov**
Fixes #

## Proposed Changes

- Update RN
- Update Gradle
- RCTAppSetupUtils.h removed from React-Core

## Checklist

- [x] I have read the rules of [**contribution**](https://github.com/haqq-network/haqq-wallet/blob/main/docs/contribution.md#important-steps).

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -86,7 +86,7 @@
     "pump": "3.0.0",
     "randombytes": "2.1.0",
     "react": "18.2.0",
-    "react-native": "0.71.12",
+    "react-native": "^0.72.4",
```
Comment (ragozin-nikita) [inline package.json:89]: Fix the version

File: package.json [hunk 2]
```diff
@@ -145,15 +145,16 @@
     "@babel/preset-typescript": "7.22.5",
     "@babel/runtime": "7.22.15",
     "@react-native-community/eslint-config": "3.2.0",
+    "@react-native/metro-config": "^0.72.11",
```
Comment (ragozin-nikita) [inline package.json:148]: Same here

File: ios/AppDelegate.swift [hunk 3]
```diff
@@ -91,41 +96,35 @@ class AppDelegate: RCTAppDelegate {
...
+
     UIView.animate(withDuration: OVERVIEW_FADE_DURATION, animations: {
       view.alpha = 0
     }) { _ in
       view.removeFromSuperview()
     }
   }
-  
-  @objc
-  func concurrentRootEnabled() -> Bool {
```
Comment (ragozin-nikita) [inline ios/AppDelegate.swift:108]: This was an unused method?

**Review by ragozin-nikita — CHANGES_REQUESTED** [review]

File: ios/AppDelegate.swift (hunk 3 above)
Comment (iGroza) [reply ios/AppDelegate.swift:108]: @devkudasov https://react-native-community.github.io/upgrade-helper/?from=0.71.12&to=0.72.4#RnDiffApp-ios-RnDiffApp-AppDelegate.mm

Comment (devkudasov) [reply ios/AppDelegate.swift:108]: > @devkudasov https://react-native-community.github.io/upgrade-helper/?from=0.71.12&to=0.72.4#RnDiffApp-ios-RnDiffApp-AppDelegate.mm

@ragozin-nikita FYI

File: package.json (hunk 1 above)
Comment (devkudasov) [reply package.json]: Version fixed

File: package.json (hunk 2 above)
Comment (devkudasov) [reply package.json]: Version fixed

**Pull request merged**